kubernetes/samples/charts/nginx/README.md

```diff
@@ -0,0 +1,209 @@
+# Install and configure NGINX
+
+## A step-by-step guide to install the NGINX operator
+See the official installation document at:
+- https://kubernetes.github.io/ingress-nginx/deploy/#using-helm
+
+As a *demonstration*, the following are steps to install the NGINX operator by using Helm 3 on a Linux OS.
+
+### 1. Add the ingress-nginx chart repository
+```
+$ helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
+$ helm repo update
+```
+Verify that the chart repository has been added.
+
+```
+$ helm search repo ingress-nginx
+NAME               CHART VERSION APP VERSION	DESCRIPTION
+ingress-nginx/ingress-nginx	2.12.0       	0.34.1     	Ingress controller for Kubernetes using NGINX a...
+```
+> **NOTE**: After updating the helm repository, the NGINX version listed maybe newer that the one appearing here, please check with the NGINX site for the lastest supported versions.
+
+### 2. Install the NGINX operator
+
+> **NOTE**: The NGINX version used for the install should match the version found with `helm search`.
+
+```
+$ kubectl create ns nginx
+$ helm install nginx-operator ingress-nginx/ingress-nginx --namespace nginx
+```
+
+Wait until the NGINX Operator is running.
+```
+$ kubectl get all -n nginx 
+NAME                                                           READY   STATUS    RESTARTS   AGE
+pod/nginx-operator-ingress-nginx-controller-84fbd64787-v4p4c   1/1     Running   0          11m
+NAME                                                        TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)                      AGE
+service/nginx-operator-ingress-nginx-controller             LoadBalancer   10.107.159.96   <pending>     80:31470/TCP,443:32465/TCP   11m
+service/nginx-operator-ingress-nginx-controller-admission   ClusterIP      10.109.12.133   <none>        443/TCP                      11m
+NAME                                                      READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/nginx-operator-ingress-nginx-controller   1/1     1            1           11m
+NAME                                                                 DESIRED   CURRENT   READY   AGE
+replicaset.apps/nginx-operator-ingress-nginx-controller-84fbd64787   1         1         1       11m
+$ POD_NAME=$(kubectl get pods -n nginx -l app.kubernetes.io/name=ingress-nginx -o jsonpath='{.items[0].metadata.name}')
+$ kubectl exec -it $POD_NAME -n nginx -- /nginx-ingress-controller --version
+
+```
+> **NOTE**: All the generated Kubernetes resources of the NGINX operator have names controlled by the NGINX Helm chart. In our case, we use `releaseName` of `nginx-operator`.
+
+## Update the NGINX operator
+After the NGINX operator is installed and running, to change some configuration of the NGINX operator, use `helm upgrade` to achieve this.
+```
+$ helm upgrade nginx-operator ingress-nginx/ingress-nginx [flags]
+```
```

Comment:
For context, can you provide an example of when to use this?  Can you note the assumptions in the command line (e.g. assumes nginx-operator is the release name)?

Reply:
Add a line 
# To update configuration on a helm chart 'nginx-operator'

Reply:
i don't know the purpose of this information
- can you provide an example of when it's needed to upgrade? 
- and can you explain the purpose of `ingress-nginx/ingress-nginx`

Reply:
ingress-nginx/ingress-nginx is the name of chart repository. 
Refer https://kubernetes.github.io/ingress-nginx/deploy/#using-helm

Upgrade is not needed in Operator Environment.  The default is sufficient.  I can remove the section to avoid the confusion.